### PR TITLE
User profile

### LINKED ISSUE #46: Create user profile page, allow rendering of purchased meals

### LINKED ISSUE #48: Make client side GET request to get users purchased
/api/users/purchases

### PR DESCRIPTION
**Pull request by vicalden23**
closes #46 
closes #48

### CODE REVIEW

File: client/src/actions/purchaseActions.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import axios from 'axios';
+
+import { attachTokenToTheHeader } from '../utils/RequestHelper';
+import { decodeToken as userDetails } from '../utils/IsAuthenticated';
+
+export const GET_PURCHASES = 'GET_PURCHASES';
+
+export function getPurchases() {
+  const user = userDetails();
+  const userId = user && user.sub;
```
Comment (obay-mardini) [inline client/src/actions/purchaseActions.js]: do we need user and userId variables in here?

File: client/src/containers/Homepage.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ class Homepage extends Component {
   };
 };
 
+
```
Comment (obay-mardini) [inline client/src/containers/Homepage.js]: i don't think this break line is important

File: client/src/containers/UserProfile.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { getPurchases } from '../actions/purchaseActions';
+import './UserProfile.css';
+
+class UserProfile extends Component {
+
+  componentDidMount() {
+
+    this.props.getPurchases();
+
+  }
```
Comment (obay-mardini) [inline client/src/containers/UserProfile.js]: can you take out the breaklines in lines 17 and 19

**Review by obay-mardini — CHANGES_REQUESTED** [review]
looks good to me except for some styling issues

**Comment by vicalden23** [comment]
👍

**Review by obay-mardini — APPROVED** [review]

File: client/src/reducers/purchaseReducer.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { GET_PURCHASES } from '../actions/purchaseActions';
```
Comment (obay-mardini) [inline client/src/reducers/purchaseReducer.js]: can you import GET_PURCHASES from '../actions' instead

**Review by obay-mardini — CHANGES_REQUESTED** [review]

**Review by obay-mardini — APPROVED** [review]

**Comment by obay-mardini** [comment]
LGTM

**Pull request merged**